client/src/components/Reviews/MetaReview.jsx

```diff
@@ -1,39 +1,63 @@
 import React, { useState, useEffect } from 'react';
-import { SummaryStar } from '../styles/reviewstyles/SummaryStar.styled.js';
 
-const MetaReview = ({ metaReview, currentProductRating }) => {
-  const [recommended, changeRecommended] = useState(0);
+import { MetaGrid, MetaRow, MetaCol, SummaryStar, Clickable } from '../styles/reviewstyles/metaReviewStyles.styled.js';
+
+const MetaReview = ({ metaReview, currentProductRating, filterRating, setFilterRating, toggleFilter, setTotalRatings, totalRatings }) => {
+  const [recommended, setRecommended] = useState(0);
+  const [ratingsArray, setRatingsArray] = useState([]);
+
 
   useEffect(() => {
     if (metaReview.recommended) {
       getRecommended(parseInt(metaReview.recommended.true), parseInt(metaReview.recommended.false));
+      getRatings();
     }
   }, [metaReview]);
 
   let getRecommended = (recommended, notRecommended) => {
-    changeRecommended(prevState => Math.round(100 * (recommended / ( recommended + notRecommended))));
+    setRecommended(Math.round(100 * (recommended / (recommended + notRecommended))));
   };
 
-  //in progress.
-  // let getRatings = () => {
-  //   let starRatings = [];
-  //   metaReview.ratings
-  // };
+  //obtains the count for each individual star rating, and the total count
+  let getRatings = () => {
+    let eachStar = Object.keys(metaReview.ratings);
+    let eachAmount = Object.values(metaReview.ratings);
+    let total = 0;
+    for (let i = 0; i < eachStar.length; i++) {
+      let starRatings = {};
+      starRatings.star = eachStar[i];
+      starRatings.amount = eachAmount[i];
+      ratingsArray.push(starRatings);
+
+      setRatingsArray(ratingsArray);
+      total += parseInt(eachAmount[i]);
+    }
+    console.log('fc', typeof setTotalRatings, 'val', totalRatings);
+    setTotalRatings(total);
+  };
 
   return (
-    <div> {metaReview.recommended ?
-      <div>
-        <h2>Average Review: {currentProductRating} < SummaryStar rating={currentProductRating} /></h2>
+    <MetaGrid> {metaReview.recommended ?
+      <MetaRow>
+        <div>< SummaryStar rating={currentProductRating} /></div>
+        <h2>Average Review: {currentProductRating}</h2>
         <div> {recommended}% of reviewers recommend this product.</div>
-        <div></div>
-        <div>5 stars <progress value = {1 / 23}></progress></div>
-        <div>4 stars</div>
-        <div>3 stars</div>
-        <div>2 stars</div>
-        <div>1 star</div>
-
-      </div> : 'Loading'}
-    </div>
+        <div>Rating Breakdown</div>
+        <div> {filterRating.length !== 0 ? <button onClick={() => setFilterRating([])}> Remove all filters</button> : null}</div>
+        <div>{[...ratingsArray].reverse().map(rating =>
+          <Clickable onClick={() => toggleFilter(parseInt(rating.star))} key = {rating + 'techdebt2'}>
```

Comment:
What is techdebt2 for? If it's something need to be done in the future, I usually add TODO in the comments

Reply:
Yes, was a temporary fix for the time being, keys have been made unique now.